### PR TITLE
feat(TPRUN-5142): client credentials for reactive authentication

### PR DESCRIPTION
**Pull request by lmeunier-talend**
**What is the problem this Pull Request is trying to solve?**
We can't use `client-credentials` with the reactive authentication 
 
**What is the chosen solution to this problem?**

- Now look for `permissions` claims while building the Authorities
- If the `X-talend-tenant-id` is on the request, set this one in the reactor context
- Fixed missing entry in the BOM for reactive-audit-logs
 
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Review by tlnd-mtallet — COMMENTED** [review]
No tests with the new `permissions` claim ? And also no test with to test extraction of tenant id from http request ?

**Comment by lmeunier-talend** [comment]
> No tests with the new `permissions` claim ? And also no test with to test extraction of tenant id from http request ?

There is one for the second case
I can add one for the first one

File: daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
 import java.util.Collection;
+import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-import org.springframework.core.convert.converter.Converter;
-import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.authority.SimpleGrantedAuthority;
-import org.springframework.security.oauth2.jwt.Jwt;
+import static ch.qos.logback.core.CoreConstants.EMPTY_STRING;
```
[the remark this answers is not in the document]
Comment (lmeunier-talend) [reply daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java]: done

**Review by tlnd-mtallet — APPROVED** [review]
LGTM

**Pull request merged**